src/parakeet/Driver.cpp

```diff
@@ -2,567 +2,4 @@
-                {
-                    printf("Checksum check failed\n");
-                    delete data;
-                }
-
-                idx += 8 + cnt * 2;
-            }
-            else
-            {
-                //3 Byte
-                if (idx + 8 + cnt * 3 > length)
-                {
-                    break;
-                }
-
-                ScanData* data = new ScanData;
-                data->from = start;
-                data->span = 360;
-                data->count = cnt;
-                unsigned short sum = start + cnt;
-
-                unsigned char* pdata = buf + idx + 6;
-                for (int i = 0; i < cnt; i++)
-                {
-                    data->intensity[i] = pdata[i * 3];
-
-                    sum += pdata[i * 3];
-
-                    unsigned short lo_byte = pdata[i * 3 + 1];
-                    unsigned short hi_byte = pdata[i * 3 + 2];
-
-                    unsigned short val = (hi_byte << 8) + lo_byte;
-
-                    sum += val;
-
-                    data->dist[i] = val;
-                }
-
-                idx += 8 + cnt * 3;
-
-                unsigned short lo = pdata[cnt * 3];
-                unsigned short hi = pdata[cnt * 3+1];
-                if (((hi<<8) | lo) == sum)
-                {
-                    onScanDataReceived(data);
-                }
-                else if(!isAutoConnecting)
-                {
-                    printf("Checksum check failed\n");
-                    delete data;
-                }
-            }
-
-            if (idx + 8 > length)
-            {
-                break;
-            }
-
-            if (buf[idx] == 0xce && buf[idx + 1] == 0xfa)
-            {
-                found = 1;
-            }
-            else
-            {
-                found = 0;
-            }
-        }
-
-        return idx;
-    }
-
-    bool Driver::sendMessageWaitForResponseOrTimeout(internal::SensorResponse::MessageType messageType, const std::string& message, std::chrono::milliseconds timeout)
-    {
-        auto startTime = std::chrono::system_clock::now();
-
-        sensorReturnMessageState[messageType] = false;
-
-        serialPort.write(message);
-
-        int msCount = 0;
-        while (!sensorReturnMessageState[messageType]
-            && std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::system_clock::now() - startTime).count() < timeout.count())
-        {
-            if (msCount % 10)
-            {
-                serialPort.write(message);
-            }
-
-            std::this_thread::sleep_for(std::chrono::milliseconds(1));
-            msCount++;
-        }
-
-        return sensorReturnMessageState[messageType];
-    }
-
-    void Driver::throwExceptionIfNotConnected()
-    {
-        if(!serialPort.isConnected())
-        {
-            throw exceptions::NotConnectedToSensorException();
-        }
-    }
-}
-}
\ No newline at end of file
+#include <parakeet/Driver.h>
```

Comment:
Seems like we don't need this file at all now?

Reply:
I didn't like the idea of removing files. Realistically, only header files should not be removed, as they can be directly interacted with. Cpp's are fine to remove. I'll get on that now.